From e63568434a516dc0ab2e5a137400b2642b45f727 Mon Sep 17 00:00:00 2001
From: Mike Achtelik <mike.achtelik@gmail.com>
Date: Wed, 20 Jan 2021 16:14:46 +0100
Subject: QQuickWindowIncubationController: Use QPointer to guard QSGRenderLoop
 reference

In some cases, when the QGuiApplication is shutting down while there
is an active QAnimationDriver and an incubating object, the
QQuickWindowIncubationController will try to access an already
destroyed QSGRenderLoop. So use a QPointer to guard the QSGRenderLoop
access.

Fixes: QTBUG-90489
Pick-to: 5.15
Pick-to: 6.0
Change-Id: I528e06ff22dfcad804593db6771d9163b21808f4
Reviewed-by: Laszlo Agocs <laszlo.agocs@qt.io>
---
 src/quick/items/qquickwindow.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git x/qtdeclarative/src/quick/items/qquickwindow.cpp y/qtdeclarative/src/quick/items/qquickwindow.cpp
index d0c9ad5454..f055814fee 100644
--- x/qtdeclarative/src/quick/items/qquickwindow.cpp
+++ y/qtdeclarative/src/quick/items/qquickwindow.cpp
@@ -164,7 +164,7 @@ protected:
 
 public slots:
     void incubate() {
-        if (incubatingObjectCount()) {
+        if (m_renderLoop && incubatingObjectCount()) {
             if (m_renderLoop->interleaveIncubation()) {
                 incubateFor(m_incubation_time);
             } else {
@@ -180,12 +180,12 @@ public slots:
 protected:
     void incubatingObjectCountChanged(int count) override
     {
-        if (count && !m_renderLoop->interleaveIncubation())
+        if (count && m_renderLoop && !m_renderLoop->interleaveIncubation())
             incubateAgain();
     }
 
 private:
-    QSGRenderLoop *m_renderLoop;
+    QPointer<QSGRenderLoop> m_renderLoop;
     int m_incubation_time;
     int m_timer;
 };
-- 
2.33.0

